permission_handler_android/lib/src/android_permission_handler_api_impls.dart

```diff
@@ -0,0 +1,135 @@
+import 'package:flutter/foundation.dart';
+import 'package:flutter/services.dart';
+import 'package:flutter_instance_manager/flutter_instance_manager.dart';
+
+import 'android_object_mirrors/activity.dart';
+import 'permission_handler.pigeon.dart';
+
+/// Host API implementation of ActivityCompat.
+class ActivityCompatHostApiImpl extends ActivityCompatHostApi {
+  /// Creates a new instance of [ActivityCompatHostApiImpl].
+  ActivityCompatHostApiImpl({
+    this.binaryMessenger,
+    InstanceManager? instanceManager,
+  })  : instanceManager = instanceManager ?? JavaObject.globalInstanceManager,
+        super(binaryMessenger: binaryMessenger);
+
+  /// Sends binary data across the Flutter platform barrier.
+  ///
+  /// If it is null, the default BinaryMessenger will be used which routes to
+  /// the host platform.
+  final BinaryMessenger? binaryMessenger;
+
+  /// Maintains instances stored to communicate with native language objects.
+  final InstanceManager instanceManager;
+
+  /// Gets whether you should show UI with rationale before requesting a permission.
+  Future<bool> shouldShowRequestPermissionRationaleFromInstance(
+    Activity activity,
+    String permission,
+  ) async {
+    final String activityInstanceId = instanceManager.getIdentifier(activity)!;
+
+    return shouldShowRequestPermissionRationale(
+      activityInstanceId,
+      permission,
+    );
+  }
+}
+
+/// Flutter API implementation of Activity.
+class ActivityFlutterApiImpl extends ActivityFlutterApi {
+  /// Constructs a new instance of [ActivityFlutterApiImpl].
+  ActivityFlutterApiImpl({
+    InstanceManager? instanceManager,
+  }) : _instanceManager = instanceManager ?? JavaObject.globalInstanceManager;
+
+  /// Maintains instances stored to communicate with native language objects.
+  final InstanceManager _instanceManager;
+
+  /// The activity currently attached to the Flutter engine.
+  ///
+  /// This is null when no activity is attached.
+  Activity? _activity;
+
+  /// Registered callbacks for activity attach events.
+  final List<void Function(Activity activity)> _onAttachedToActivityCallbacks =
+      [];
+
+  /// Registered callbacks for activity detach events.
+  final List<void Function()> _onDetachedFromActivityCallbacks = [];
+
+  /// Adds a callback to be called when an activity is attached.
+  ///
+  /// If an activity is attached when this method is called, the callback is
+  /// called immediately.
+  void addOnAttachedToActivityCallback(
+    void Function(Activity attachedActivity) onActivityAttached,
+  ) {
+    if (_activity != null) onActivityAttached(_activity!);
```

Comment:
According to the Google Flutter convention `if` statements should always have braces, the one exception is when you directly return.

```suggestion
    if (_activity != null) { 
      onActivityAttached(_activity!);
    }
```